Tools/GMSInfo/source/main.cpp

```diff
@@ -7,262 +7,45 @@
+    std::string typesDataBaseFilePath;
+
+    if (!(cli({"-L", "\"--level\""}) >> levelArchivePath))
     {
-        printf("ERROR: You need to specify input and output files!\n\tUsage: gmstool.exe <GMS File> <PRM File> [Path to new uncompressed GMS]\n");
-        return 0;
+        spdlog::error("You must present path to the level archive!");
+        return -1;
     }
 
-    const char* inFile    = argv[1];
-    const char* prmInFile = argv[2];
-
-    printf(" *** GMS TOOL ***\n");
-    printf(" * IN  : %s\n", inFile);
+    cli("--types", "typeids.json") >> typesDataBaseFilePath;
 
-    FILE* fp = fopen(inFile, "rb");
-    if (!fp)
+    if (!ReGlacier::TypesDataBase::GetInstance().Load(typesDataBaseFilePath))
     {
-        printf("Failed to open file %s\n", inFile);
-        return -400;
+        spdlog::error("Failed to load types database from file {}", typesDataBaseFilePath);
+        return -2;
     }
 
-    fseek(fp, 0L, SEEK_END);
-    auto rawBufferSize = ftell(fp);
-    rewind(fp);
-
-    char* rawBuffer = (char*)malloc(rawBufferSize);
-    auto readyBytes = fread(rawBuffer, sizeof(char), rawBufferSize, fp);
-
-    if (rawBufferSize != readyBytes)
+    // Open level archive
+    auto level = std::make_unique<ReGlacier::LevelDescription>(levelArchivePath);
+    if (!level->Open())
     {
-        printf("Failed to read full contents from raw stream! Required %d ready %d\n", rawBufferSize, readyBytes);
-        fclose(fp);
+        spdlog::error("Failed to open level {}", levelArchivePath);
         return -1;
     }
-    fclose(fp);
-
-    // decompress
-    GMS gms = { 0 };
-    gms.field_0 = 1;
-    gms.m_raw = (int)rawBuffer;
-
-    int v5 = *(unsigned long*)rawBuffer;
-
-    gms.field_C = v5;
-    gms.field_8 = *(unsigned long*)(rawBuffer + 4);
-    gms.field_14 = (*(unsigned char*)(rawBuffer + 8)) != 0;
-
-    int outBuffSize = (v5 + 15) & 0xFFFFFFF0;
-    char* outBuffer = (char*)malloc(outBuffSize);
-    memset(outBuffer, 0x0, outBuffSize);
-    GMS_Decompress(&gms, outBuffer, outBuffSize);
-
-    /*
-    if (argv > 3) {
-        const char* outFile = argv[3];
-        printf(" * OUT : %s\n", outFile);
-
-        printf(" *** DUMP ***\n");
-        printf(" Decompressed size: 0x%.8X (%d)\n", gms.field_C, gms.field_C);
-        for (int i = 0; i < outBuffSize; i += 4)
-        {
-            printf("%.4X %.2X %.2X %.2X %.2X\n", i, outBuffer[i], outBuffer[i + 1], outBuffer[i + 2], outBuffer[i + 3]);
-        }
-
-        FILE* outFP = fopen(outFile, "a+");
-        if (outFP)
-        {
-            fwrite(outBuffer, sizeof(char), outBuffSize, outFP);
-            fclose(outFP);
-        }
-        else
-        {
-            printf("Failed to open file %s to write result!\n", outFile);
-        }
-
-        printf(" --- PARSING DONE ---\n");
-    }
-    */
-
-    loadTypesDataBase();
-    analyzeGMS(prmInFile, outBuffer, outBuffSize);
 
-    free(rawBuffer);
-    free(outBuffer);
+    level->AnalyzeGMS();
```

Comment:
add more options:
 * analyze GMS
 * generate report about GMS
 * save uncompressed GMS to file (and compress back our GMS)